From: Valentine Fatiev <valentinef@nvidia.com>
Subject: [PATCH] BACKPORT:
 drivers/net/ethernet/mellanox/mlx5/core/diag/en_rep_tracepoint.h

Change-Id: I9b3ca16a1c3a751cd913894a5d94e82ff3f280fd
---
 .../net/ethernet/mellanox/mlx5/core/diag/en_rep_tracepoint.h  | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/ethernet/mellanox/mlx5/core/diag/en_rep_tracepoint.h
+++ b/drivers/net/ethernet/mellanox/mlx5/core/diag/en_rep_tracepoint.h
@@ -25,7 +25,11 @@ TRACE_EVENT(mlx5e_rep_neigh_update,
 			struct in6_addr *pin6;
 			__be32 *p32;
 
+#ifdef HAVE_ASSIGN_STR_1_PARAM
 			__assign_str(devname);
+#else
+ 			__assign_str(devname, nhe->neigh_dev->name);
+#endif
 			__entry->neigh_connected = neigh_connected;
 			memcpy(__entry->ha, ha, ETH_ALEN);
 
